src/thingsboard.c

```diff
@@ -189,6 +192,68 @@ static void client_request_time(struct k_work *work)
 	k_work_reschedule(k_work_delayable_from_work(work), K_SECONDS(10));
 }
 
+static int client_handle_rpc_response(struct coap_client_request *req, struct coap_packet *response)
+{
+	const uint8_t *payload;
+	uint16_t payload_len;
+	uint8_t code;
+	char code_str[5];
+	char expected_code_str[5];
+
+	LOG_INF("%s", __func__);
+
+	code = coap_header_get_code(response);
+	if (code != COAP_RESPONSE_CODE_CONTENT) {
+		coap_response_code_to_str(code, code_str);
+		coap_response_code_to_str(COAP_RESPONSE_CODE_CONTENT, expected_code_str);
+		LOG_ERR("Unexpected response code for RPC request: got %s, expected %s", code_str,
+			expected_code_str);
+		return -1;
+	}
+
+	payload = coap_packet_get_payload(response, &payload_len);
+	if (!payload_len) {
+		LOG_ERR("Received an empty RCP response");
+		return payload_len;
+	}
+
+	if (rpc_cb) {
+		rpc_cb(payload, payload_len);
+	}
+
+	return 0;
+}
+
+void thingsboard_rpc(const char *method, rpc_callback_t cb, ...)
+{
+	int err;
+	va_list param;
+	char params[64];
```

Comment:
We of course don't know how long the parameters list will be but 64 characters seems very short to me. Maybe increase it a bit to 256 or something like that?

Reply:
yeah, you're right, this was the first thing I used for testing, it might well be too short. I will increase it